File: 6kyu/squared_spiral_1/complete_solution.py
def squared_spiral(n):
    x_dist, y_dist = 1, 1
    x_min, y_min = -1, -1
    x_max, y_max = 1, 1
    direction = "east"
    counter = 0
    x, y = 0, 0

    while counter < n:
        if direction == "east":
            x += 1

            if x == x_max:
                direction = "north"
                y_max = y + y_dist
                x_dist += 1

        elif direction == "north":
            y += 1

            if y == y_max:
                direction = "west"
                x_min = x - x_dist
                y_dist += 1

        elif direction == "west":
            x -= 1

            if x == x_min:
                direction = "south"
                y_min = y - y_dist
                x_dist += 1

        elif direction == "south":
            y -= 1

            if y == y_min:
                direction = "east"
                x_max = x + x_dist
                y_dist += 1

        counter += 1

    return (x, y)

File: 6kyu/squared_spiral_2/c++/complete_solution.cpp
int squaredSpiral(int x, int y)
{
    int xDist = 1;
    int yDist = 1;
    int xMin = -1;
    int yMin = -1;
    int xMax = 1;
    int yMax = 1;

    int xPos = 0;
    int yPos = 0;

    char direction = 'e';
    int counter = 0;

    while ((x != xPos) || (y != yPos))
    {
        if (direction == 'e')
        {
            xPos++;

            if (xPos == xMax)
            {
                direction = 'n';
                yMax = yPos + yDist;
                xDist++;
            }
        }
        else if (direction == 'n')
        {
            yPos++;

            if (yPos == yMax)
            {
                direction = 'w';
                xMin = xPos - xDist;
                yDist++;
            }
        }
        else if (direction == 'w')
        {
            xPos--;

            if (xPos == xMin)
            {
                direction = 's';
                yMin = yPos - yDist;
                xDist++;
            }
        }
        else if (direction == 's')
        {
            yPos--;

            if (yPos == yMin)
            {
                direction = 'e';
                xMax = xPos + xDist;
                yDist++;
            }
        }

        counter++;
    }

    return counter;
}